com.qq.QQmusic.yml

```diff
@@ -0,0 +1,70 @@
+app-id: com.qq.QQmusic
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: /app/bin/qqmusic.sh
+separate-locales: false
+
+finish-args:
+    - '--share=ipc'
+    - '--share=network'
+    - '--socket=x11'
+    - '--socket=session-bus'
+    - '--socket=pulseaudio'
+    - '--device=dri'
+    - '--device=shm'
+    - '--filesystem=/tmp'   # for singleton
+
+build-options:
+    strip: false
+    no-debuginfo: true
+
+modules:
+    -
+        name: qqmusic
+        only-arches:
+            - x86_64
+        sources:
+            -
+                type: script
+                commands:
+                    - /app/extra/QQmusic/qqmusic --no-sandbox
+                dest-filename: qqmusic.sh
+            -
+                type: script
+                commands:
+                    - ar x qqmusic_amd64.deb
+                    - tar -xf data.tar.xz
+                    - mv opt/QQmusic .
+                    - rm -rf opt usr control.tar.gz data.tar.xz debian-binary
+                dest-filename: apply_extra
+            -
+                type: file
+                path: com.qq.QQmusic.desktop
+            -
+                type: file
+                path: com.qq.QQmusic.png
+            -
+                type: file
+                path: com.qq.QQmusic.appdata.xml
+            -
+                type: extra-data
+                filename: qqmusic_amd64.deb
+                url: http://dldir1.qq.com/music/clntupate/linux/deb/qqmusic_1.0.6_amd64.deb
+                sha256: 42e0663a5439286e64adfc035bcbb82731548bab25f5e628fb085b050a5926c2
+                size: 74527272
+                x-checker-data:
+                    type: html
+                    url: https://y.qq.com/download/download.html
+                    version-pattern: http://dldir1.qq.com/music/clntupate/linux/deb/qqmusic_(\S*)_amd64.deb
+                    url-pattern: http://dldir1.qq.com/music/clntupate/linux/deb/qqmusic_(\S*)_amd64.deb
+        buildsystem: simple
+        build-commands:
+            - install -D -m 755 -t /app/bin/ qqmusic.sh
+            - install -D -m 755 -t /app/bin/ apply_extra
+            - install -D -m 644 -t /app/share/icons/hicolor/256x256/apps/ com.qq.QQmusic.png
+            - install -D -m 644 -t /app/share/applications/ com.qq.QQmusic.desktop
+            - install -D -m 644 -t /app/share/appdata/ com.qq.QQmusic.appdata.xml
```

Comment:
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html